### PR TITLE
documentation: first draft of release notes

### PR DESCRIPTION
**Pull request by lazan**
A draft PR to make it easier for us to produce the final release-notes. We need to do this together.

### CODE REVIEW

File: documentation/release-notes.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+### runtime configuration 
+
+`casual` can now update all parts of configuration while running. No part is 
+restarted.
+
+A convenient way for a user to update configuration is to runtime edit.
+
+```bash
+$ casual configuration --edit
```
Comment (ahhud) [inline documentation/release-notes.md:16]: A central change is also what you implies here that "configuration" is moved from "subsystem" _domain_ to "subsystem" _configuration_. Even if it's not _adding_ many new features and rather moves them, I think it is worth mentioning.

Comment (lazan) [reply documentation/release-notes.md:16]: I concur.

File: documentation/release-notes.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+### branch on inbound
+
+In `1.7` we branch a transaction on the way 'in' to the domain. This change is
+primarily because it gives technical advantages, but it could also benefit 
+business flows. It is more likely to access a resource with the same branch in 
+complex topologies. Sort of a poor mans _tightly coupled_ transactions.
+
+### gateway pair of ipc - tcp sockets
+
+`inbound` and `outbound` tcp connections now has it's own _ipc device_ associated.
```
Comment (AlbsAgren) [inline documentation/release-notes.md]:
```suggestion
`inbound` and `outbound` tcp connections now have their own _ipc devices_ associated.
```

**Review by AlbsAgren — APPROVED** [review]

File: documentation/release-notes.md [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+
+When a service is called that has a timeout, a deadline is set based on the timeout.
+This deadline is propagated downstream to consecutively called services.
+If a service is invoked with an ongoing deadline, the earliest of the ongoing and 
+the timeout based deadline is set for the service. 
+
+Therefore, the first service sets an _execution window_ to which downstream service 
+invocations will conform.
+
+One, perhaps negative, consequence of this semantic is that services downstream
```
Comment (ahhud) [inline documentation/release-notes.md:33]: We are using strictly objective terms and then lists a possible negative consequence, shouldn't we balance this off with something positive that explain why we did this change? :-)

Comment (lazan) [reply documentation/release-notes.md:33]: Check. Do a _suggestion_ how you want it?

File: documentation/release-notes.md [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+`casual transaction --scale-instances`    | `casual transaction --scale-resource-proxies`
+
+
+* see [configuration.operation.md](../middleware/administration/documentation/cli/configuration.operation.md)
+
+
+removed                             |  use
+------------------------------------|------------------------------
+`casual gateway --list-services`    | `casual service --list-instances`
+`casual gateway --list-queues`      | `casual queue --list-instances`
```
Comment (ahhud) [inline documentation/release-notes.md]: This is not correct, see issue #459 .

**Pull request merged**